src/Cassandra/Mapping/TypeConversion/TypeConverter.cs

```diff
@@ -96,6 +96,12 @@ protected TypeConverter()
         /// </summary>
         internal object ConvertToDbFromUdtFieldValue(Type valueType, Type dbType, object value)
         {
+            var nullableType = typeof(Nullable<>).MakeGenericType(dbType);
+            if (valueType == nullableType)
+            {
+                return value == null ? null : nullableType.GetTypeInfo().GetProperty("Value").GetValue(value);
```

Comment:
You can use just `return value` for the nullable type regardless of the underlying value.